lib/internal/tls.js

```diff
@@ -305,15 +305,15 @@ function configSecureContext(context, options = {}, name = 'options') {
     }
   }
 
-  if (clientCertEngine !== undefined) {
+  if (typeof clientCertEngine === 'string') {
     if (typeof context.setClientCertEngine !== 'function')
       throw new ERR_CRYPTO_CUSTOM_ENGINE_NOT_SUPPORTED();
-    if (typeof clientCertEngine !== 'string') {
-      throw new ERR_INVALID_ARG_TYPE(`${name}.clientCertEngine`,
-                                     ['string', 'null', 'undefined'],
-                                     clientCertEngine);
-    }
-    context.setClientCertEngine(clientCertEngine);
+    else
+      context.setClientCertEngine(clientCertEngine);
+  } else if (clientCertEngine !== undefined) {
```

Comment:
Do we need to add a test for this? I'd guess that this code is not currently covered by any tests, but I could be wrong.

Reply:
Oh, wait, I'm wrong (I think)! Ignore.